### PR TITLE
[Core] add clone method to python interface of properties

### PR DESCRIPTION
**Pull request by armingeiser**
@KratosMultiphysics/interface-committee 

I have added a clone function to the python interface of a property. Like this one can create a copy of a property at python level.

### CODE REVIEW

File: kratos/python/add_properties_to_python.cpp
Comment (philbucher) [inline kratos/python/add_properties_to_python.cpp]: Why don’t you call clone?

I would say this creates another ref to the same properties rather than cloning it
Have you tested if it actually does not modify the existing one when you modify the new one?

Why don’t you call Clone? Or in case there is no clone I would say you should call the copyCtor

Comment (armingeiser) [reply kratos/python/add_properties_to_python.cpp]: There is no clone, and here i am calling the copy constructor (line111). But as it has to be a shared pointer. thats why i can not create the object on the stack directly by the copy constructor.

Comment (armingeiser) [reply kratos/python/add_properties_to_python.cpp]: I can of course also make a clone method at the properties object, but i thought this is less intrusive.

**Comment by loumalouomega** [comment]
Can we first merge https://github.com/KratosMultiphysics/Kratos/pull/2919? , I am tired of fix that PR, and @philbucher, @RiccardoRossi is waiting for your review

**Comment by philbucher** [comment]
> Can we first merge #2919? , I am tired of fix that PR, and @philbucher, @RiccardoRossi is waiting for your review

I agree, then we can speak in person next week @armingeiser this is easier than writing, I hope it is not urgent.

@loumalouomega I know, I will do it Monday latest when I have access to my computer again, sorry for the delay

**Comment by armingeiser** [comment]
Ok for me, its not urgent and i can write a test in the meantime.

File: kratos/python/add_properties_to_python.cpp
Comment (philbucher) [inline kratos/python/add_properties_to_python.cpp]: Yep thats also what I had in mind 👍 

Although I am checking now the pybind docs if there is a native solution

Comment (philbucher) [reply kratos/python/add_properties_to_python.cpp]: Seems like it should be possible to expose the copy constructor
Then the py and cpp implementation would be consistent

Comment (armingeiser) [reply kratos/python/add_properties_to_python.cpp]: But how could python know when to use the copy constructor and when to just create a "reference" - because we need to avoid that python makes copies whenever we write 
```python
property2 = property1
```
that is not pythonic...

Comment (armingeiser) [reply kratos/python/add_properties_to_python.cpp]: Or do you mean 
```python
property2 = KM.Properties(property1)
```

**Review by philbucher — COMMENTED** [review]
To me this looks good, I will approve after the subproperies are merged

**Comment by rubenzorrilla** [comment]
Allowing this makes it possible to have two property objects with same Id. Am I wrong?

What would happen if someone tries to add both properties to the same model part?

**Comment by armingeiser** [comment]
@rubenzorrilla you are right. This creates a copy of the property, with the same id and more important all the property values.
If it is added to the same modelpart, an error is thrown that you can not add another property with the same id. If it should be added to the modelpart afterwards, one can change the id of the new property before doing so.

The same would happen if a property is created "from scratch" with the same id and then added to the same model part.

**Comment by rubenzorrilla** [comment]
> @rubenzorrilla you are right. This creates a copy of the property, with the same id and more important all the property values.
> If it is added to the same modelpart, an error is thrown that you can not add another property with the same id. If it should be added to the modelpart afterwards, one can change the id of the new property before doing so.
> 
> The same would happen if a property is created "from scratch" with the same id and then added to the same model part.

You're right (I didn't remember about the error thrown). More than this, my point was if we should allow properties to be copied. Considering that the c++ implementation has a copy constructor which is not deleted, makes sense to add such clone method to Python.

(sorry for the noise)

**Comment by armingeiser** [comment]
#2919 is merged, can we proceed with this?

**Review by philbucher — APPROVED** [review]

**Pull request merged**
